jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -856,7 +859,8 @@ public Session getSession(boolean create)
         @Override
         public Request getWrapped()
         {
-            return (Request)super.getWrapped();
+            // Identical to (Request)super.getWrapped() except that the cast is costly here.
+            return request;
```

Comment:
We have similar code in 
- RetainableByteBuffer
- ConnectionMetaData
- ServerUpgradeRequestDelegate
- ServerUpgradeResponseDelegate

Of these, maybe only RetainableByteBuffer may have any impact, but we should remove the anti-pattern from our code and avoid casting our wrapped types
-

Reply:
It is a tradeoff between memory (we don't want to store the same reference in N fields, one of each wrapper), and this JDK issue, and so far it has only surfaced for `Request`, so I would not bother for the others.

Reply:
Can we at least look at the usage of retainedbytebuffer wrapper, to see if applies at least a little bit

Reply:
Before generalizing this pattern, there's a 3rd contender on the list reported by the type pollution agent that should be looked at.

I did a quick analysis, and that 3rd one isn't going to be easy, as IMHO its root lies in a design mistake we made. Very briefly, the `ExecutionStrategy.Producer` should have specified `Invocable.Task` instead of plain `Runnable`.

Reply:
@sbordet FYI I analyzed the layout of `Request.Wrapper` in multiple conditions and found out that this extra field is costing 8 bytes in all conditions, making the heap consumption grow from 16 to 24 with G1 and Shenandoah and from 24 to 32 with Z because of compressed oops, padding and memory alignment.

Clearly, we're trading off memory for speed here.

Reply:
Note that it might be better to simply not use the Attributes.Wrapper base class.  I do not think there is any situation that Request.Wrapper  must be an Attributes.Wrapper.  It can just implement the Attributes delegate methods itself.  I think that is a small cost to pay for a space saving on every single request.

Reply:
I changed `Request.Wrapper` so it doesn't extend from `Attributes.Wrapper` anymore. This gives us the best of both worlds perf wise (i.e.: the speed boost and the best memory efficiency) in exchange of the minuscule usability cost of not being an `Attributes.Wrapper` and not being able to use `Attributes.unwrap` anymore.